samples/transfer-workflow/src/main/java/com/example/transfer/application/ImporterWorkflow.java

```diff
@@ -0,0 +1,118 @@
+package com.example.transfer.application;
+
+import akka.Done;
+import akka.javasdk.annotations.ComponentId;
+import akka.javasdk.client.ComponentClient;
+import akka.javasdk.workflow.Workflow;
+import akka.javasdk.workflow.WorkflowContext;
+import akka.stream.Materializer;
+import com.example.transfer.application.ImporterWorkflowBackgroundProcess.MyBackgroundCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static akka.Done.done;
+
+@ComponentId("importer")
+public class ImporterWorkflow extends Workflow<ImporterWorkflow.ImporterWorkflowState> {
+
+  private static final Logger log = LoggerFactory.getLogger(ImporterWorkflow.class);
+  private final Materializer materializer;
+  private final ComponentClient componentClient;
+  private final String importerId;
+
+  public ImporterWorkflow(Materializer materializer, ComponentClient componentClient, WorkflowContext context) {
+    this.materializer = materializer;
+    this.componentClient = componentClient;
+    this.importerId = context.workflowId();
+  }
+
+  @Override
+  public Optional<BackgroundProcess<ImporterWorkflowState>> backgroundProcess() {
+
+    return Optional.of(new ImporterWorkflowBackgroundProcess(componentClient, importerId, materializer));
+  }
+
+  public record ImporterWorkflowState(boolean started, int offset, boolean finished) {
+    public ImporterWorkflowState withOffset(int offset) {
+      return new ImporterWorkflowState(started, offset, finished);
+    }
+
+    public ImporterWorkflowState start() {
+      return new ImporterWorkflowState(true, offset, false);
+    }
+
+    public ImporterWorkflowState stop() {
+      return new ImporterWorkflowState(true, offset, true);
+    }
+  }
+
+  @Override
+  public ImporterWorkflowState emptyState() {
+    return new ImporterWorkflowState(false, 0, false);
+  }
+
+  @Override
+  public WorkflowDef<ImporterWorkflowState> definition() {
+    var startImport = step("start-import")
```

Comment:
Using the step to start the workflow, once started it will be restarted by the lifecycle in case of rebalancing.

Reply:
I have added this comment before, but it seems that GH didn't register it. 

I don't think this will work. The `asyncCall` will be marked as executer and it won't be re-executed. We will move to the transition directly.